File: 1652-defuse-the-bomb/1652-defuse-the-bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        int n=code.size();
        
        int sIndex=1, eIndex=k;
        
        if(k < 0)
            k=-1*k, sIndex=n-k, eIndex=n-1;
        
        int windowSum=0;
        
        for(int i=sIndex; i<=eIndex; i++) 
        {
            windowSum+=code[i];
        }
        
        vector<int> res(n);
        
        for(int i=0; i<n; i++)
        {
            res[i]=windowSum;
            windowSum-=code[(sIndex++)%n];
            windowSum+=code[(++eIndex)%n];
        }
        
        return res;        
    }
};

File: 1652-defuse-the-bomb/1652-defuse-the-bomb.java
class Solution {
    public int[] decrypt(int[] code, int k) {
        int n = code.length;
        
        int[] ans = new int[n];
        
        if(k == 0) {
            return ans;
        }
        
        int sum = 0;
        int start = 1, end = k;
        
        if(k < 0) {
            start = n - Math.abs(k);
            end = n - 1;
        }
        
        for(int i = start; i <= end; i++) {
            sum += code[i];
        }
        
        for(int i = 0; i < n; i++) {
            ans[i] = sum;
            
            sum -= code[start % n];
            sum += code[(end + 1) % n];
            
            start++;
            end++;
        }
        
        return ans;
    }
}